### PR TITLE
Removing obsolete stuff and moving tests to NUnit3

### PR DESCRIPTION
**Pull request by alexeyzimarev**
- Removing obsolete platforms:
   - .NET 3.5-4.5.1 - the earliest is 4.5.2
   - Silverlight
   - Windows Phone (both of them)
   - MonoTouch
- Updated NUnit to v3
- Fixed tests and build script to use NUnit3
- Fixed async tests to actually use async/await

### CODE REVIEW

File: RestSharp.Tests/JsonTests.cs [hunk 1]
```diff
@@ -1060,7 +1062,7 @@ public string CreateDynamicJsonStringDictionary()
 
         private static T GetPayLoad<T>(string fileName)
         {
-            string doc = File.ReadAllText(Path.Combine("SampleData", fileName));
+            string doc = File.ReadAllText(Path.Combine(currentPath, "SampleData", fileName));
```
Comment (mjwsteenbergen) [inline RestSharp.Tests/JsonTests.cs]: Why is this change neccesary?

Comment (alexeyzimarev) [reply RestSharp.Tests/JsonTests.cs]: NUnit3 console runner does not guarantee to execute in the test library location. As a result, none of the test files are found by tests. You can try yourself or resolve it differently, I find this way the most efficient.

Comment (mjwsteenbergen) [reply RestSharp.Tests/JsonTests.cs]: 👍

**Comment by mjwsteenbergen** [comment]
No, it won't ;)

**Comment by alexeyzimarev** [comment]
Thanks! Bold move.

**Pull request merged**